src/main/java/org/randywebb/wlts/beans/AbstractBean.java

```diff
@@ -19,319 +19,331 @@
-	/** Get a String of all the standard String fields in a comma separated list of key = value.
-		@return Standard string values as a string.
-	*/
-	@Override
-	public String toString() {
-		String value = "";
-
-		for (HashMap.Entry<String,String> entry : entrySet()) {
-			value += (value.length() == 0 ? "" : ", ") + entry.getKey() + " = " + entry.getValue();
-		}
-
-		return value;
-	}
+public abstract class AbstractBean extends HashMap<String, String> {
+
+    /** log any errors or warnings. */
+    private static Logger log = LoggerFactory.getLogger(AbstractBean.class);
+
+    /** Field names which are Boolean. See forCSV() */
+    private List<String> booleanFields;
+    /** Field names which are Date. See forCSV() */
+    private List<String> dateFields;
+    /** Field names which are Integer. See forCSV() */
+    private List<String> integerFields;
+    /** Format for Date fields. See forCSV() */
+    private SimpleDateFormat dateFormat;
+
+    /** Initialize the default. */
+    public AbstractBean() {
+        this.booleanFields = new ArrayList<String>();
+        this.dateFields = new ArrayList<String>();
+        this.integerFields = new ArrayList<String>();
+    }
+
+    /** Initialize with fields that could be Date, Boolean, or Integer.
+        @param dateFrmt Format to use to generate and parse Date strings
+        @param booleanFieldList The names of the fields that are Booleans
+        @param integerFieldList The names of the fields which are Integers
+        @param dateFieldList The names of the fields which are Dates
+    */
+    public AbstractBean(SimpleDateFormat dateFrmt, List<String> booleanFieldList, List<String> integerFieldList, List<String> dateFieldList) {
+        dateFormat = dateFrmt;
+        booleanFields = null == booleanFieldList ? new ArrayList<String>() : booleanFieldList;
+        dateFields = null == dateFieldList ? new ArrayList<String>() : dateFieldList;
+        integerFields = null == integerFieldList ? new ArrayList<String>() : integerFieldList;
+    }
+
+    /** Helper for subclasses to convert JSON Array to List.
+        See setFromJSON() for how subclasses can handle non-string values in the JSON
+        @param <T> The type of the subclass
+        @param array The JSON Array to parse
+        @param toFill The List to fill with parsed objects
+        @param clazz The Subclass.class. Would be T.class, if Java allowed that.
+        @return List of subclass instances from the JSON array
+    */
+    protected static <T> List<T> fromArray(JSONArray array, List<T> toFill, Class<T> clazz) {
+
+        if (null != array) {
+            try {
+                Constructor<T> constructor = clazz.getConstructor(JSONObject.class);
+
+                for (Object o : array) {
+                    try {
+                        toFill.add(constructor.newInstance(new Object[] {(JSONObject) o}));
+                    } catch (InvocationTargetException e) {
+                        if (log.isErrorEnabled()) {
```

Comment:
I usually only surround trace, debug, and info messages with the if block.
The extra boolean check isn't really necessary here as the code will only execute when there is an exception. I guess you'd save a bit of performance if the log is set to FATAL instead of ERROR, but it's not something that impacts the "normal flow" of the program.
It is OK to have it, but generally not necessary for anything > INFO level